node-manager/src/node.rs

```diff
@@ -28,13 +29,135 @@ impl Node {
         let keys_manager = create_keys_manager(mnemonic, node_index.child_index);
         let pubkey = pubkey_from_keys_manager(&keys_manager);
 
-        let tcp_proxy = Arc::new(TcpProxy::new().await);
-
         Ok(Node {
             uuid: node_index.uuid,
             pubkey,
             keys_manager: Arc::new(keys_manager),
-            tcp_proxy,
         })
     }
+
+    pub async fn connect_peer(
+        &self,
+        websocket_proxy_addr: String,
+        peer_pubkey_and_ip_addr: String,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        if peer_pubkey_and_ip_addr.is_empty() {
+            info!(
+                "ERROR: connectpeer requires peer connection info: `connectpeer pubkey@host:port`"
+            );
+            return Err("connectpeer requires peer connection info".into());
+        };
+        let (pubkey, peer_addr) = match parse_peer_info(peer_pubkey_and_ip_addr) {
+            Ok(info) => info,
+            Err(e) => {
+                info!("ERROR: could not parse peer info: {}", e);
+                return Err(e.into());
+            }
+        };
+
+        if connect_peer_if_necessary(websocket_proxy_addr, pubkey, peer_addr)
+            .await
+            .is_ok()
+        {
+            info!("SUCCESS: connected to peer: {}", pubkey);
+            Ok(())
+        } else {
+            info!("ERROR: could not connect to peer: {}", pubkey);
+
+            Err("could not connect to peer".into())
+        }
+    }
+}
+
+pub(crate) async fn connect_peer_if_necessary(
+    websocket_proxy_addr: String,
+    _pubkey: PublicKey,
+    peer_addr: SocketAddr,
+    // peer_manager: Arc<PeerManager>,
+) -> Result<(), ()> {
+    // TODO add this when the peer manager is ready
+    /*
+    for node_pubkey in peer_manager.get_peer_node_ids() {
+        if node_pubkey == pubkey {
+            return Ok(());
+        }
+    }
+    */
+
+    // first make a connection to the node
+    let tcp_proxy = TcpProxy::new(websocket_proxy_addr, peer_addr).await;
+
+    // TODO remove the test send
+    tcp_proxy.send();
+
+    // TODO then give that connection to the peer manager
+
+    // TODO then schedule a reader on the connection
+
+    Ok(())
+}
+
+pub(crate) fn parse_peer_info(
+    peer_pubkey_and_ip_addr: String,
+) -> Result<(PublicKey, SocketAddr), std::io::Error> {
```

Comment:
Is there no helper for this in ldk?

Reply:
Also the Err type should be MutinyError

Reply:
No, this function comes from LDK-sample in fact, since LDK doesn't have any helpers for it. Unless they added it and have not updated the ldk-sample either. 

Changed it to `MutinyError`